### ISSUE TITLE
Cannot find module 'next/dist/server/next-server.js'. Turborepo + VercelCLI

### ISSUE THREAD

**Issue by MrDockal** [vercel/next.js#48949]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #76~20.04.1-Ubuntu SMP Mon Mar 20 15:54:19 UTC 2023
    Binaries:
      Node: 16.15.1
      npm: 8.19.1
      Yarn: 1.22.19
      pnpm: 7.29.1
    Relevant packages:
      next: 13.3.1
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0

----------------- 

Vercel CLI 28.20.0
28.20.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Turbopack (--turbo)

### Link to the code that reproduces this issue

https://github.com/MrDockal/nextjs-turbo-vercel-cli-issue

### To Reproduce

1. Clone the repo  https://github.com/MrDockal/nextjs-turbo-vercel-cli-issue
2. npx vercel build --token YOUR_VERCEL_TOKEN --cwd apps/web
3. npx vercel deploy --prebuilt --token YOUR_VERCEL_TOKEN --yes --cwd apps/web
4. Visit the `/lambda` page and you'll get 500. The root page `/` works

### Describe the Bug

Hello,
When I deploy a lambda function (page with getServerSideProps implemented) to Vercel using **Vercel CLI + Turborepo + NextJs**. The lambda throws an error: _**Cannot find module 'next/dist/server/next-server.js'**_ (it could similar to this issue https://github.com/vercel/next.js/discussions/39432)
There isn't **node_modules** folder at all
![image](https://user-images.githubusercontent.com/8297716/235138170-7365f8a0-5008-4719-985e-687fd99031c9.png)

I tried to deploy the same Nextjs app outside the Turborepo and it works. As you see there is **node_modules** folder with all required dependencies now
![image](https://user-images.githubusercontent.com/8297716/235138489-ae12123f-a26e-431f-922e-5257415e9dfc.png)

I don't know if this is an issue with Next, Turborepo or Vercel CLI itself.
Thank you

### Expected Behavior

I expect endpoint `/lambda` to return 200 with HTML code

**Comment by padmaia** [comment]
Removed the Turbopack label since it's related to production builds, which Turbopack does not support yet.

**Comment by jakeleventhal** [comment]
Related issue: https://github.com/vercel/vercel/issues/8794

**Comment by jakeleventhal** [comment]
https://github.com/vercel/next.js/discussions/39432#discussioncomment-6080594
Check this out. Turbo unrelated

**Comment by MrDockal** [comment]
Right! thank you @jakeleventhal. This is definitely related to https://github.com/trappar/next-vercel-ssr-bug

**Comment by joseocabarcas** [comment]
I have the same issue with turborepo and pnpm when deploy with _vercel cli prebuilt_. The pages with `getServerSideProps` broken with _`Cannot find module 'next/dist/server/[next-server](https://linear.app/vercel/issue/NEXT-server).js'`_
Any solutions or workaround?

**Comment by jakeleventhal** [comment]
Should close this as a duplicate of https://github.com/vercel/vercel/issues/8794
